src/components/QuickGrid/QuickGrid.tsx

```diff
@@ -142,25 +143,40 @@ export class QuickGridInner extends React.Component<IQuickGridProps, IQuickGridS
             this.columnsMinTotalWidth = columnsToDisplay.map(x => x.minWidth || defaultMinColumnWidth).reduce((a, b) => a + b, 0);
         }
     }
+
+    componentWillUpdate(nextProps, nextState) {
+        this.finalGridRows = getRowsSelector(nextState, nextProps);
+    }
+
     componentDidUpdate(prevProps, prevState) {
-        this.finalGridRows = getRowsSelector(this.state, this.props);
         if (prevProps.columns !== this.props.columns || prevState.groupBy !== this.state.groupBy || prevState.columnWidths !== this.state.columnWidths) {
             this._grid.recomputeGridSize();
         } else if (this.state.sortDirection !== prevState.sortDirection || this.state.sortColumn !== prevState.sortColumn) {
             this._grid.forceUpdate();
         }
     }
 
-    getGridWidth() {
+
+    getViewportWidth() {
```

Comment:
Like the name change to Viewport.

Reply:
Just a step in calculating GridWidth, getGridWidth is using it now.